### ISSUE TITLE
Mouse pointer not visible on vAmiga 0.9.9

### ISSUE THREAD

**Issue by pedgarcia** [dirkwhoffmann/vAmiga#399]
Hi

I was testing vAmiga 0.9.9 with KS 3.1 rev 40.63 and WB 3.1. Everything seems to work and load fine, however, when I click the emulator Window to get the mouse pointer "captured" I can't see it at all on the Amiga screen.

I know it is there somewhere because I can randomly click things but the mouse pointer is not visible at all. Have you encountered that issue before?

I am using Mac mini (Late 2012) with macOS 10.15.15 (Catalina).

Thanks

**Comment by shinra-electric** [comment]
This could be related to #350?  
I had something similar when playing Monkey Island. 
Try using KickStart 1.3 rather than 3.1

**Comment by dirkwhoffmann** [comment]
> This could be related to #350?

Yes, this might well be related. We didn't check the Kick 3.1 / Monkey Island configuration in UAE, didn't we? 😬

Since Kick 3.1 / WB 3.1 works well in UAE (with ECS Agnus / OCS Denise), I guess Monkey Island works, too, in UAE.

**Comment by dirkwhoffmann** [comment]
This is an interesting bug, because everything looks alright in the DMA debugger (Sprite DMA is happening)...

<img width="1018" alt="Bildschirmfoto 2020-08-05 um 08 39 38" src="https://user-images.githubusercontent.com/12561945/89381604-baf46100-d6f9-11ea-9723-102ce401bc4f.png">

**Comment by shinra-electric** [comment]
I just checked now. 

Yes, the cursor displays properly in FS-UAE with an A500 and v3.1KS.

![Screenshot 2020-08-05 at 14 58 50](https://user-images.githubusercontent.com/50119606/89387226-64544a80-d72c-11ea-9aaf-fad883cc06f2.png)

**Comment by dirkwhoffmann** [comment]
Coming close. All sprite DMA is intact and even the inspector is able to display the sprite. But as you can see, the horizontal coordinate gets lost. This means the sprite is displayed at pixel position 2 which is outside the visible area. I think I already know what it is... stay tuned...

<img width="980" alt="Bildschirmfoto 2020-08-05 um 10 27 08" src="https://user-images.githubusercontent.com/12561945/89397435-22b5a680-d710-11ea-8235-90691a901e9d.png">

**Comment by dirkwhoffmann** [comment]
Fixed in v0.9.10.1.

<img width="1018" alt="Bildschirmfoto 2020-08-05 um 12 00 42" src="https://user-images.githubusercontent.com/12561945/89399853-56de9680-d713-11ea-8be9-a76a226942b3.png">

Here is what happened: When a sprite register gets changed, vAmiga records the change in a buffer. At the end of a raster line, the sprite drawing code replays that buffer and synthesizes sprite pixels. However, the sprite drawing code wasn't called inside the VBLANK area, because no sprites can be displayed there. Hence, if a sprite register was modified inside the VBLANK area, the register change wasn't recognized. It most cases, it still worked coincidentally, because the register records were not deleted at the end of a raster line and were replayed later (at their original horizontal coordinates which is wrong).

I've already checked if this bug fixes one of the other open issues, but it doesn't.
